src/OpenApi/src/Extensions/JsonTypeInfoExtensions.cs

```diff
@@ -11,6 +11,21 @@ namespace Microsoft.AspNetCore.OpenApi;
 
 internal static class JsonTypeInfoExtensions
 {
+    private static readonly List<Type> _exemptPrimitives =
```

Comment:
Are these from an OpenAPI spec? Otherwise there are probably other primitives, e.g. `nint`, `byte`

Reply:
Added these and a few more other primitive types. Lemme know what you think of the list.

Reply:
Any of `Guid`, `Uri`, `TimeSpan`, `DateOnly` and `TimeOnly`? Just comparing to the list we have in Swashbuckle and they're the only ones that are different to this set. They basically end up as a `string` with a `format` of some kind in the schema.

Reply:
Ah, yes! I recall you sharing this in the other PR. I've updated this set so it's consistent between the two. I wasn't able to find the logic in NSwag for this but this probably a set that we should be consist with throughout the implementations. 🤔

Reply:
I'd forgotten about `Version` 😄

Did you want `TimeSpan` too, or was there a reason to skip that?

Reply:
Yet another reason to make this public 😓 Added that as well. Let me know if there is anything else I missed. I'd like to get this PR in soon to ease some of the build failures it is causing but happy to add things in a follow-up PR.

Reply:
The new set seems to cover all the common primitives/cases I'm aware of 👍